### PR TITLE
SSDT DMAC - FWHD - B0D4

### PR DESCRIPTION
**Pull request by Devil-66**
in new intel generations, dmac and fwhd are missing.

### CODE REVIEW

**Comment by Devil-66** [comment]
seee foto before
before

and after b0d4 return zero
after

**Comment by asgmeonerandom** [comment]
which intel generations? what is it trying to fix?

File: Docs/AcpiSamples/SSDT-B0D4.dsl [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * Only necessary when B0D4 device is present in the (INTEL) DSDT.
+ * For all intel generations, This Configuration Adjust B0D4 device Return (Fix DSDT Warnings and error).
+ */
+DefinitionBlock("", "SSDT", 2, "ACDT", "B0D4", 0x00000000)
+{
+    External (_SB_.PCI0.B0D4, DeviceObj)
+    
+    Scope (_SB.PCI0.B0D4)
+    {    
+        If (_OSI ("Darwin"))
```
Comment (vit9696) [inline Docs/AcpiSamples/SSDT-B0D4.dsl:11]: Some method is missing here.

File: Docs/AcpiSamples/SSDT-B0D4.dsl [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/*
+ * Only necessary when B0D4 device is present in the (INTEL) DSDT.
+ * For all intel generations, This Configuration Adjust B0D4 device Return (Fix DSDT Warnings and error).
```
Comment (vit9696) [inline Docs/AcpiSamples/SSDT-B0D4.dsl:3]: I still cannot make sense what this SSDT does from the description.

File: Docs/AcpiSamples/SSDT-DMAC.dsl [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/*
+ * Only necessary when no DMAC device (with any name) is present in the DSDT and a custom device-id
```
Comment (vit9696) [inline Docs/AcpiSamples/SSDT-DMAC.dsl:2]: Why is that device needed? Also not very descriptive to be honest.

File: Docs/AcpiSamples/SSDT-FWHD.dsl [hunk 4]
```diff
@@ -0,0 +1,35 @@
+/*
+ * Only necessary when no FWHD device (with any name) is present in the DSDT and a custom device-id
```
Comment (vit9696) [inline Docs/AcpiSamples/SSDT-FWHD.dsl:2]: Same as DMAC.

**Comment by vit9696** [comment]
This needs more background information on what is being done. Both in the PR description and in the dsl tables themselves.

File: Docs/AcpiSamples/SSDT-B0D4.dsl (hunk 1 above)
Comment (Devil-66) [reply Docs/AcpiSamples/SSDT-B0D4.dsl:11]: there is no lack of method, it is just so. see photo above

File: Docs/AcpiSamples/SSDT-DMAC.dsl (hunk 3 above)
Comment (Devil-66) [reply Docs/AcpiSamples/SSDT-DMAC.dsl:2]: the descriptions need to be changed, I was waiting for your comment to change them

Comment (Devil-66) [reply Docs/AcpiSamples/SSDT-DMAC.dsl:2]: dmac e fwhd, these devices are missing on dsdt from 6 gen intel onwards often, and on real mac there are.

File: Docs/AcpiSamples/SSDT-B0D4.dsl (hunk 1 above)
Comment (vit9696) [reply Docs/AcpiSamples/SSDT-B0D4.dsl:11]: Then how does it work? You cannot return a value from just the scope, it is not a function. Read the ACPI spec.

**Comment by foskvs** [comment]
I guess somebody doesn’t even bother reading APCI speciifications and doesn’t have a clue on coding syntax...

**Comment by Devil-66** [comment]
> This needs more background information on what is being done. Both in the PR description and in the dsl tables themselves.

ok for dmac ssdt and fwhd, which are missing on new intel generations, but apple uses them on new intel generations

**Comment by Tandem91** [comment]
Next step? Could you also work on SSDT_SMC? It would help to avoid using a SMC emulator.

**Comment by vit9696** [comment]
It is unjustified why these devices are needed on our machines. Just the fact they are present in Apple ACPI does not make it a requirement for our ACPI. Closing until more information is available.

**Comment by dreamwhite** [comment]
> It is unjustified why these devices are needed on our machines. Just the fact they are present in Apple ACPI does not make it a requirement for our ACPI. Closing until more information is available.

This should be said also to Mald0n tbh ._. but hey "APPLE USES THOSE DEVICES SO WE SHOULD INCLUDE THEM WITHOUT ANY KIND OF DOCUMENTATION". Pretty ridiculous

**Comment by Download-Fritz** [comment]
@foskvs @dreamwhite Please refrain from making any comments that do not detail anything about the PR's content, this is not a gossip forum.

@Tandem91 That is not how ACPI works.

**Pull request closed without merging**